src/khoj/database/adapters/__init__.py

```diff
@@ -394,20 +394,41 @@ async def aget_client_application_by_id(client_id: str, client_secret: str):
 
 
 class AgentAdapters:
-    DEFAULT_AGENT_NAME = "khoj"
+    DEFAULT_AGENT_NAME = "Khoj"
     DEFAULT_AGENT_AVATAR = "https://khoj-web-bucket.s3.amazonaws.com/lamp-128.png"
+    DEFAULT_AGENT_SLUG = "khoj"
 
     @staticmethod
     async def aget_agent_by_id(agent_id: int, user: KhojUser):
         agent = await Agent.objects.filter(id=agent_id).afirst()
-        # Check if it's accessible to the user
+        if agent and (agent.public or agent.creator == user):
+            return agent
+        return None
+
+    @staticmethod
+    async def aget_agent_by_slug(agent_slug: str):
+        return await Agent.objects.filter(slug__iexact=agent_slug.lower()).afirst()
```

Comment:
Should this helper method also check if retrieved agents by slug are public or created by user?